x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/classification_exploration/column_data.tsx

```diff
@@ -4,79 +4,101 @@
 
 interface ColumnData {
   actual_class: string;
   actual_class_doc_count: number;
-  predicted_class?: string;
-  count?: number;
-  error_count?: number;
+  [key: string]: any;
 }
 
 export const ACTUAL_CLASS_ID = 'actual_class';
+export const OTHER_CLASS_ID = 'other';
+export const MAX_COLUMNS = 6;
 
 export function getColumnData(confusionMatrixData: ConfusionMatrix[]) {
   const colData: Partial<ColumnData[]> = [];
+  const columns: Array<{ id: string; display?: any }> = [
+    {
+      id: ACTUAL_CLASS_ID,
+      display: <span />,
+    },
+  ];
 
-  confusionMatrixData.forEach((classData: any) => {
-    const correctlyPredictedClass = classData.predicted_classes.find(
-      (pc: PredictedClass) => pc.predicted_class === classData.actual_class
-    );
-    const incorrectlyPredictedClass = classData.predicted_classes.find(
-      (pc: PredictedClass) => pc.predicted_class !== classData.actual_class
-    );
+  let showOther = false;
 
-    let accuracy;
-    if (correctlyPredictedClass !== undefined) {
-      accuracy = correctlyPredictedClass.count / classData.actual_class_doc_count;
-      // round to 2 decimal places without converting to string;
-      accuracy = Math.round(accuracy * 100) / 100;
-    }
+  confusionMatrixData.forEach(classData => {
+    const otherCount = classData.other_predicted_class_doc_count;
 
-    let error;
-    if (incorrectlyPredictedClass !== undefined) {
-      error = incorrectlyPredictedClass.count / classData.actual_class_doc_count;
-      error = Math.round(error * 100) / 100;
+    if (otherCount > 0) {
+      showOther = true;
     }
 
-    let col: any = {
+    const col: any = {
       actual_class: classData.actual_class,
       actual_class_doc_count: classData.actual_class_doc_count,
+      other: otherCount,
     };
 
-    if (correctlyPredictedClass !== undefined) {
-      col = {
-        ...col,
-        predicted_class: correctlyPredictedClass.predicted_class,
-        [correctlyPredictedClass.predicted_class]: accuracy,
-        count: correctlyPredictedClass.count,
-        accuracy,
-      };
-    }
+    const predictedClasses = classData.predicted_classes || [];
 
-    if (incorrectlyPredictedClass !== undefined) {
-      col = {
-        ...col,
-        [incorrectlyPredictedClass.predicted_class]: error,
-        error_count: incorrectlyPredictedClass.count,
-      };
+    columns.push({ id: classData.actual_class });
+
+    for (let i = 0; i < predictedClasses.length; i++) {
+      const predictedClass = predictedClasses[i].predicted_class;
+      const predictedClassCount = predictedClasses[i].count;
+      col[predictedClass] = predictedClassCount;
     }
 
     colData.push(col);
   });
 
-  const columns: any = [
+  if (showOther) {
+    columns.push({ id: OTHER_CLASS_ID });
+  }
+
+  return { columns, columnData: colData };
+}
+
+export function getTrailingControlColumns(numColumns: number, setShowFullColumns: any) {
```

Comment:
You could import `EuiDataGridControlColumn` from EUI and add as a return type like `export function getTrailingControlColumns(numColumns: number, setShowFullColumns: any): EuiDataGridControlColumn[] {`. IMHO we should do that at least in a follow up, I'm fine getting the PR with this like it is for FF. Eventually this change will allow us to get rid of the `@ts-ignore` in `evaluate_panel.tsx` for the data grid's `trailingControlColumns` attribute.

Reply:
Added in 422c7aacdfc9d129a7877af2d9f1c8ebede15f80